File: 3254-find-the-power-of-k-size-subarrays-i/3254-find-the-power-of-k-size-subarrays-i.py
class Solution:
    def resultsArray(self, nums: List[int], k: int) -> List[int]:
        if k == 1:
            return nums
        l = []
        bad = 0
        for i in range(len(nums)):
            if i:
                bad += nums[i] != nums[i-1] + 1
            if i >= k:
                bad -= nums[i - k] + 1 != nums[i - k + 1]
            if i >= k - 1:
                l.append(nums[i] if bad == 0 else -1)
        return l

File: 3254-find-the-power-of-k-size-subarrays-i/3254-find-the-power-of-k-size-subarrays-i.cpp
class Solution {
public:
    vector<int> resultsArray(vector<int>& nums, int k){
        int n=nums.size();
        vector<int> res;
        for(int i=k-1;i<n;i++){
            vector<int> ca;
            for(int j=0;j<k;j++){
                ca.push_back(nums[i-(k-1)+j]);
            }
            int w=ca.back();
            for(int i=1;i<ca.size();i++){
                if(ca[i-1]+1!=ca[i]){w=-1; break;}
            }
            res.push_back(w);
        }
        return res;
    }
};